lib/Elastica/Search.php

```diff
@@ -74,9 +81,11 @@ class Search
      * Constructs search object.
      *
      * @param \Elastica\Client $client Client object
+     * @param BuilderInterface $builder
      */
-    public function __construct(Client $client)
+    public function __construct(Client $client, BuilderInterface $builder = null)
     {
+        $this->_builder = $builder ?: new Builder();
```

Comment:
As above, I think it would be better for the Client to know nothing about the Builder.